### DISCUSSION TITLE
Customize charts similar to Ggplot2 and Tableau

### DISCUSSION THREAD

**Discussion by cmajorros** [apache/superset#29700] [Ideas]
Hi folks, 
As a user, there is some pain points that I faced while using superset especially when I need to create charts an do some customization such as removing grid line , applying the color , adjust the axis , adding and formatting secondary axis   In addition, some charts even though they are similar but the ability might be different such as a bar cannot applying forecasting , but another charts can do.  I have been using Tableau and GGplot2 in R (also available in Python) for a long time. I think the way to create chart of the 2 tools quite similar and more flexible comparing with using pre-built chart. Is there any possibility to use GGplot2 in Superset. User will just select geometry (such as line, bar, dot, shape type, text as they need), size (based on measurement), coloring (base or a dimension or measurement) , axis (can select scale type such as logarithm, ordinal scale etc) , can on - off grid line and so on.  That would be fantastic if we could do that. In case we could just write python code base on the existing dataset then show the picture of the chart in superset. That seems work for me as well.

**Comment by sfirke** [comment]
Hello, long time ggplot2 user here so I feel your pain that Superset charts are much less customizable.  I am far from an expert on the technical side of this but you should know that Superset uses the Apache ECharts library for its charting.  Any such customization would have to be compatible with ECharts.  For instance, sometimes users ask if they can change the color of the font on the chart axes with CSS; they can't, because that happens in ECharts.

I would love to see something like Superset but with the flexibility of ggplot2 ... but I expect that would require such a big rewrite that it would pretty much be a new application at that point.
